### ISSUE TITLE
Viewport-Texture repeat flag has no effect

### ISSUE THREAD

**Issue by Gulum** [godotengine/godot#36820]
**Godot version:**
3.2 stable
**OS/device including version:**
Windows 10 1903
**Issue description:**
Changing Viewport-Texture-Flag of Spatial-Material albedo to repeat has no effect. In contrast, changing filter mode works.

**Steps to reproduce:**
1. Create fresh project.
2. Define viewport with something rendered to it
3. Apply viewport-texture to some surface and check repeat flag. (make sure resource is local to scene)
4. Change UV scaling to something larger then 1 to see the effect.

**Minimal reproduction project:**
[Bug.zip](https://github.com/godotengine/godot/files/4293315/Bug.zip)

**Comment by Gulum** [comment]
Image for ref: 
![image](https://user-images.githubusercontent.com/7810662/75990064-9ac73880-5ef4-11ea-8406-e0ad1ae27909.png)

**Comment by clayjohn** [comment]
I think Viewport texture flags need to be changed in GDScript for them to have any effect.

**Comment by Gulum** [comment]
I already tried to set them via a tool-gdscript but no luck this time. The weird thing is that the filter flag in the editor works as supposed whereas the repeat flag does nothing.

**Comment by Lexpartizan** [comment]
Yes, I encountered this error and killed half a day until I found this message and realized what was wrong))

**Comment by TheDuriel** [comment]
Comes with the fact that to work, these flags actually require the texture to be reimported.

**Comment by Lexpartizan** [comment]
In this case, we need to look for a workaround. At the same time, it is productive. Why not just make repeat behavior for default viewports?

**Comment by TheDuriel** [comment]
The flags should have not been exposed originally. This behavior will change in 4.0, where they will be set on the shader level, not the texture itself.

For now, adjusting the material shader to repeat the texture via UV manipulation will be the way forward.

**Comment by Shadowblitz16** [comment]
I really need a way to wrap tile maps in a parallax-layer is there any workaround for this?

**Comment by Lexpartizan** [comment]
@Shadowblitz16 try vec2 uv=fract (UV) or uv=UV-floor(UV).
This may lead to artifacts, but it will still work.

**Comment by navy-cap-echo** [comment]
Hello
@Calinou I see that we are at the version 3.4.beta 5 and unfortunately this issue is still not resolved.
Should I give up all hope ?

**Comment by Calinou** [comment]
> Hello @Calinou I see that we are at the version 3.4.beta 5 and unfortunately this issue is still not resolved. Should I give up all hope ?

As far as I know, nobody has found a solution for this in `3.x`. It's possible that this issue will be resolved in 4.0 with the Vulkan renderer rewrite still (not tested yet).
